keep/providers/sumologic_provider/sumologic_provider.py

```diff
@@ -0,0 +1,439 @@
+            )
+        else:
+            raise Exception(str(monitor_update_response.json()))
+
+    def setup_webhook(
+        self, tenant_id: str, keep_api_url: str, api_key: str, setup_alerts: bool = True
+    ):
+        try:
+            parsed_url = urlparse(keep_api_url)
+
+            # Extract the query string
+            query_params = parsed_url.query
+
+            # Find the provider_id in the query parameters
+            # connection_template.json is the payload that will be sent to keep as an event
+            provider_id = query_params.split("provider_id=")[-1]
+            connection_name = f"KeepHQ-{provider_id}"
+            connection_payload = {
+                "type": "WebhookDefinition",
+                "name": connection_name,
+                "description": "A webhook connection that pushes alerts to KeepHQ",
+                "url": keep_api_url,
+                "headers": [],
+                "customHeaders": [{"name": "X-API-KEY", "value": api_key}],
+                "defaultPayload": open(
+                    rf"{Path(__file__).parent}/connection_template.json"
+                ).read(),
+                "webhookType": "Webhook",
+                "connectionSubtype": "Event",
+                "resolutionPayload": open(
+                    rf"{Path(__file__).parent}/connection_template.json"
+                ).read(),
+            }
+            # Creating a sumo logic connection
+            connection_id = self.__create_connection(
+                connection_payload=connection_payload, connection_name=connection_name
+            )
+
+            # Monitors
+            monitors = self.__get_monitors_without_keep(connection_id=connection_id)
+
+            # Install connections in monitors that don't have keep
+            for monitor in monitors:
+                self.__install_connection_in_monitor(
+                    monitor=monitor, connection_id=connection_id
+                )
+        except Exception as e:
+            raise e
+
+    @staticmethod
+    def __extract_severity(severity: str):
+        if "critical" in severity.lower():
+            return AlertSeverity.CRITICAL
+        elif "warning" in severity.lower():
+            return AlertSeverity.WARNING
+        elif "missing" in severity.lower():
+            return AlertSeverity.INFO
+
+    @staticmethod
+    def __extract_status(status: str):
+        if "resolved" in status.lower():
+            return AlertStatus.RESOLVED
+        else:
+            return AlertStatus.FIRING
+
+    @staticmethod
+    def _format_alert(
+        event: dict,
+        provider_instance: Optional["SumologicProvider"] = None,
+    ) -> AlertDto:
+        return AlertDto(
+            id=event["id"],
+            name=event["name"],
+            severity=SumologicProvider.__extract_severity(
+                severity=event["triggerType"]
+            ),
+            fingerprint=event["id"],
```

Comment:
I'm worried that id is not right for fingerprint. If the alert gets updated (e.g., status changes), is it grouped correctly?

Reply:
![image](https://github.com/user-attachments/assets/8b5f8b7a-1103-4cfe-8144-09c64825d0a5)

Reply:
Id sent is the ID of the monitor that triggered the alert.